podpac/core/algorithm/algorithm.py

```diff
@@ -130,6 +130,9 @@ def f(node):
                 inputs[key] = node.eval(coordinates, output=output, _selector=_selector)
             self._multi_threaded = False
 
+        if settings["ALGORITHM_XARRAY_FLOATING_POINT_CORRECTION"]:
```

Comment:
I wouldn't do this SOLELY through a global setting. It's best if individual nodes can over-ride this behavior if needed. 

```suggestion
        if self.xarray_floating_point_correction is None:
             if settings["ALGORITHM_XARRAY_FLOATING_POINT_CORRECTION"]:
                  inputs = align_xarray_dict(inputs)
       elif self.xarray_floating_point_correction:
```

Reply:
You'll have to add the traited attribute `xarray_floating_point_correction` to the algorithm class as well.

Reply:
I put the trait in the Algorithm class and it defaults to the global settings value. Will this cause issues with settings load behavior? Would the settings every get loaded after this trait is used in normal operations?

Reply:
Much better fix than my suggestion. I think that's actually what we typically do in PODPAC.